plugins/woocommerce/src/Internal/Admin/Orders/PageController.php

```diff
@@ -255,21 +257,103 @@ private function set_action(): void {
 	 * @return void
 	 */
 	public function register_menu(): void {
-		$order_types = wc_get_order_types( 'admin-menu' );
+		$order_types    = wc_get_order_types( 'admin-menu' );
+		$has_shop_order = in_array( 'shop_order', $order_types, true );
+		$hook_mappings  = array();
+
+		if ( $has_shop_order ) {
+			// Process shop_order first to create top-level menu.
+			$post_type = get_post_type_object( 'shop_order' );
+			$menu_name = $post_type->labels->menu_name;
+
+			/**
+			 * Filters whether to include the processing order count in the menu.
+			 *
+			 * @since 2.5.0
+			 * @param bool $include_count Whether to include the count. Default true.
+			 */
+			if ( apply_filters( 'woocommerce_include_processing_order_count_in_menu', true ) && current_user_can( 'edit_others_shop_orders' ) ) {
```

Comment:
Seems odd to have registration of menus be conditional to this filter?

Reply:
Also the code that's been added here seems to be repeating much of the logic in `menu_order_count`?